src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -5066,52 +5066,199 @@ class ForClauseSegment(BaseSegment):
     )
 
 
+class ExecuteOptionSegment(BaseSegment):
+    """An option for EXEC/EXECUTE WITH clause."""
+
+    type = "execute_option"
+
+    _result_sets_definition = OneOf(
+        # ( { column_name data_type [ COLLATE collation_name ]
+        # [ NULL | NOT NULL ] } [,...n ] )
+        Bracketed(
+            Delimited(
+                Sequence(
+                    Ref("ColumnReferenceSegment"),
+                    Ref("DatatypeSegment"),
+                    Sequence(
+                        "COLLATE",
+                        Ref("ObjectReferenceSegment"),
+                        optional=True,
+                    ),
+                    OneOf("NULL", Sequence("NOT", "NULL"), optional=True),
+                ),
+            )
+        ),
+        # AS OBJECT [ db_name . [ schema_name ] . | schema_name . ]
+        # {table_name | view_name | table_valued_function_name }
+        Sequence(
+            "AS",
+            "OBJECT",
+            Sequence(
+                Ref("SingleIdentifierGrammar", optional=True),
+                Ref("SingleIdentifierGrammar"),
+                optional=True,
+            ),
+            Ref("ObjectReferenceSegment"),
+        ),
+        # AS TYPE [ schema_name.]table_type_name
+        Sequence(
+            "AS",
+            "TYPE",
+            Sequence(
+                Ref("ObjectReferenceSegment"),
+                Ref("DotSegment"),
+                optional=True,
+            ),
+            Ref("ObjectReferenceSegment"),
+        ),
+        # AS FOR XML
+        Sequence("AS", "FOR", "XML"),
+    )
+
+    match_grammar = OneOf(
+        "RECOMPILE",
+        Sequence("RESULT", "SETS", "UNDEFINED"),
```

Comment:
nit, this could be cleaner as one sequence ... "RESULT", "SETS", OneOf(...)

Reply:
Thanks @WittierDinosaur  for the review!

The original spec is the following:
```sql
<execute_option>::=
{
        RECOMPILE
    | { RESULT SETS UNDEFINED }
    | { RESULT SETS NONE }
    | { RESULT SETS ( <result_sets_definition> [,...n ] ) }
}
```

So it can be a standalone `RECOMPILE`, or it can one of the three variations beginning with `RESULT SETS` ... Do you mean to create another `OneOf()` under the first `OneOf()`?